### PR TITLE
Chapter 11 & 12 fixes

### PR DESCRIPTION
**Pull request by szantopeter**
Fix issues in localstack and AWS for chapter 11 and 12. See the commit messages for the list of issues

### CODE REVIEW

File: chapters/chapter-11/cdk/src/main/java/dev/stratospheric/todoapp/cdk/ServiceApp.java [hunk 1]
```diff
@@ -35,15 +35,13 @@ public static void main(final String[] args) {
...
-    String dockerImageTag = (String) app.getNode().tryGetContext("dockerImageTag");
-    requireNonEmpty(dockerImageTag, "context variable 'dockerImageTag' must not be null");
-
     String region = (String) app.getNode().tryGetContext("region");
     requireNonEmpty(region, "context variable 'region' must not be null");
 
+    String dockerImageUrl = (String) app.getNode().tryGetContext("dockerImageUrl");
+    requireNonEmpty(dockerImageUrl, "context variable 'dockerImageUrl' must not be null");
+
+
```
Comment (BjoernKW) [inline chapters/chapter-11/cdk/src/main/java/dev/stratospheric/todoapp/cdk/ServiceApp.java:44]:
```suggestion
```

File: chapters/chapter-12/application/build.gradle [hunk 2]
```diff
@@ -28,14 +28,14 @@ dependencies {
...
   implementation 'io.awspring.cloud:spring-cloud-aws-starter-sqs'
+  implementation 'org.flywaydb:flyway-core'
   implementation "org.thymeleaf.extras:thymeleaf-extras-springsecurity6:${thymeleafExtrasSpringSecurityVersion}"
   implementation 'software.amazon.awssdk:cognitoidentityprovider'
   implementation 'com.google.code.findbugs:jsr305:3.0.2'
   implementation 'org.webjars:webjars-locator-core'
   implementation 'nz.net.ultraq.thymeleaf:thymeleaf-layout-dialect'
   implementation 'org.webjars:bootstrap:4.6.1'
   implementation 'org.webjars:font-awesome:5.15.3'
-  developmentOnly 'org.springframework.boot:spring-boot-devtools'
```
Comment (BjoernKW) [inline chapters/chapter-12/application/build.gradle:38]: What's the exact issue here? We should still be able to use the Spring Boot DevTools.

**Review by BjoernKW — COMMENTED** [review]
Looks good in general. I just have a question regarding the removal of the Spring Boot DevTools.

File: chapters/chapter-12/application/src/test/resources/localstack/init-ses.sh [hunk 3]
```diff
@@ -1,11 +1,9 @@
...
-awslocal sqs create-queue --queue-name stratospheric-todo-sharing
-
 awslocal ses verify-email-identity --email-address noreply@stratospheric.dev
 awslocal ses verify-email-identity --email-address info@stratospheric.dev
 awslocal ses verify-email-identity --email-address tom@stratospheric.dev
 awslocal ses verify-email-identity --email-address bjoern@stratospheric.dev
 awslocal ses verify-email-identity --email-address philip@stratospheric.dev
 
-echo "Initialized."
+echo "SES Setup complete."
```
Comment (giuliopulina) [inline chapters/chapter-12/application/src/test/resources/localstack/init-ses.sh:9]: I haven't tried, but I think that this change would make [AbstractDevIntegrationTest](https://github.com/stratospheric-dev/stratospheric/blob/4b64c791910e884883e1c39500d59fdcccd19388/application/src/test/java/dev/stratospheric/todoapp/AbstractDevIntegrationTest.java#L28) fail

**Comment by BjoernKW** [comment]
Closing this pull request for now, since there are still open questions and failing tests.

**Comment by szantopeter** [comment]
Hi @BjoernKW the question was open, because I answered your question and I was waiting for you to accept it my answer. There weren't any failing test either.

**Comment by BjoernKW** [comment]
> Hi @BjoernKW the question was open, because I answered your question and I was waiting for you to accept it my answer. There weren't any failing test either.

Please see this comment of mine as well as @giuliopulina comment about a failing test:

"What's the exact issue here? We should still be able to use the Spring Boot DevTools."

**Comment by szantopeter** [comment]
Hi @BjoernKW I answered @giuliopulina 's question too. I think it is based on a misunderstanding. I attach a screen shot of my earlier answer here 
image

**Comment by giuliopulina** [comment]
> Hi @BjoernKW I answered @giuliopulina 's question too. I think it is based on a misunderstanding. I attach a screen shot of my earlier answer here image



> Hi @BjoernKW I answered @giuliopulina 's question too. I think it is based on a misunderstanding. I attach a screen shot of my earlier answer here image

> Hi @BjoernKW I answered @giuliopulina 's question too. I think it is based on a misunderstanding. I attach a screen shot of my earlier answer here image

You are right, the integration test AbstractDevIntegrationTest is only on the main application and not on the chapters. Nevertheless, I think that some consistency should be kept (if the chapters' folders are meant to be the steps needed to build the application, I would expect to have the same code between chapters and main application, wherever possible).

**Comment by szantopeter** [comment]
fair point, if @BjoernKW is willing to reopen this PR and prefers this change, then I am happy to apply these changes to the main application too. One thing is for sure, the sample applications in the chapters are broken for now

**Comment by BjoernKW** [comment]
> fair point, if @BjoernKW is willing to reopen this PR and prefers this change, then I am happy to apply these changes to the main application too. One thing is for sure, the sample applications in the chapters are broken for now

As @rieckpil has mentioned before, continuing to update the "chapter" applications and keeping them all in sync ultimately would be a combinatorial maintenance nightmare. These applications are to be understood as educational stepping stones while the only authoritative - and always up-to-date - application is the main one

**Comment by szantopeter** [comment]
@BjoernKW your comment makes sense, I am just not sure what is your intent with the chapter applications that are broken now for multiple reasons and this PR would fix them.

**Pull request closed without merging**